awesome_sso/service/depends.py

```diff
@@ -45,17 +47,58 @@ async def sso_registration(
     return RegisterModel(**payload)
 
 
-async def sso_user_email(payload: dict = Depends(sso_token_decode)) -> EmailStr:
+def sso_user_email(payload: dict = Depends(sso_token_decode)) -> EmailStr:
     if "email" not in payload:
-        logger.warn(payload)
+        logger.warning(payload)
         raise NotFound("email not found")
     return payload["email"]
 
 
-async def sso_user(user_email: dict = Depends(sso_user_email)) -> AwesomeUserType:
+async def sso_user(user_email: EmailStr = Depends(sso_user_email)) -> AwesomeUserType:
     user = await Settings[AwesomeUserType]().user_model.find_one(
         Settings[AwesomeUserType]().user_model.email == user_email
     )
     if user is None:
         raise NotFound("user not found")
     return user
+
+
+async def jwt_token_decode(sso: str = Cookie(None)) -> JWTPayload:
+    try:
+        payload = jwt.decode(
+            sso, Settings.public_key, algorithms=[ASYMMETRIC_ALGORITHM]
+        )
+    except Exception as e:
+        environment = os.environ.get("ENV")
+        if environment is None or environment == "":
+            logger.warning("using mock user")
+            payload = {"user_id": MOCK_USER_ID}
+        else:
+            logger.warning(e)
+            raise Unauthorized(str(e))
+
+    try:
+        jwt_payload = JWTPayload(sso_user_id=PydanticObjectId(payload["user_id"]))
+    except Exception as e:
+        raise Unauthorized(str(e))
+    return jwt_payload
+
+
+def sso_user_id(
+    payload: JWTPayload = Depends(jwt_token_decode),
+) -> PydanticObjectId:
+    if payload.sso_user_id is None:
+        logger.warning(payload)
+        raise NotFound("sso user id not found")
```

Comment:
no that I think of it, I do think a 422 UnprocessableEntity or 400 BadRequest is more suitable in this case.